editor/plugins/node_3d_editor_plugin.cpp

```diff
@@ -6200,9 +6235,24 @@ void Node3DEditor::_finish_grid() {
 	}
 }
 
+bool setup = false;
+Vector3 cumulative_movement = Vector3();
+Vector3 last_position = Vector3();
```

Comment:
I would move those to be members of `Node3DEditor`. We can also simplify the movement check logic by simply storing one `Vector3` with the position where the grid was last updated, and checking the distance against that.

Reply:
sounds good yeah I'll do that

Reply:
done